libs/blocks/global-navigation/global-navigation.js

```diff
@@ -710,21 +714,23 @@ class Gnav {
   };
 }
 
-export default async function init(header) {
-  const { locale } = getConfig();
-  // TODO locale.contentRoot is not the fallback we want if we implement centralized content
-  const url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
-  const resp = await fetch(`${url}.plain.html`);
-  const html = await resp.text();
-  if (!html) return null;
-  const parsedHTML = await replaceText(html, getFedsPlaceholderConfig(), undefined, 'feds');
-  processMartechAttributeMetadata(parsedHTML);
-
+export default async function init(block) {
   try {
-    const gnav = new Gnav(new DOMParser().parseFromString(parsedHTML, 'text/html').body, header);
+    const { locale } = getConfig();
+    const url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
+    const content = await fetchAndProcess({
+      url,
+      e: 'Error fetching gnav content',
+    });
+    if (!content) return null;
+    const gnav = new Gnav({
+      content,
+      block,
+      useFederatedContent: url.includes('/federal/'),
```

Comment:
It's not obvious if we actually have a need for `useFederatedContent` anymore, does it serve any purpose?

Reply:
I'm unsure about one menu use-case, but I'll check. We might be able to completely get rid of this now as I've restructured the code around the central `fetchAndProcess(PlainHtml)` method

Reply:
Yeah, we should be good to get rid of it completely - since we federatePictureSources right as we fetch anything